agent/connect/ca/provider_vault_auth_jwt.go

```diff
@@ -0,0 +1,73 @@
+package ca
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+func NewJwtAuthClient(authMethod *structs.VaultAuthMethod) (*VaultAuthClient, error) {
+	params := authMethod.Params
+
+	role, ok := params["role"].(string)
+	if !ok || strings.TrimSpace(role) == "" {
+		return nil, fmt.Errorf("missing 'role' value")
+	}
+
+	// The path is required for the auto-auth config, but this auth provider
+	// seems to be used for jwt based auth by directly passing the jwt token.
+	// So we only require the token file path if the token string isn't
+	// present.
+	needTokenPath := true
+	// support legacy setup that allows directly passing the `jwt`
+	if _, ok := hasJWT(params); ok {
+		needTokenPath = false
+	}
+	if needTokenPath {
```

Comment:
nit / not-blocking: the `needTokenPath` could be removed, I think?

```suggestion
	// support legacy setup that allows directly passing the `jwt`
	if _, ok := hasJWT(params); !ok {
```

Reply:
I was seriously thinking about reworking this completely anyways as it could use the same legacyCheck() used in the Azure PR to do something similar. Having them all use the same method would be good even if it were very simple in this case. With the review for that (azure) PR completed I think I'll get it merged so I can rework this code to use it and clean this up. Thanks.

Reply:
Rework of this pushed up. Going to resolve this thread as the code in question is gone.